### PR TITLE
Currently deserves to be framed.

### PR DESCRIPTION
**Pull request by Raenore**
Basically this is based on the "NPC Speech" frame idea.

There's a new toolbar button using the "ui_chat" interface icon, which can be toggled in "Frames settings".
image
image

Clicking on it will reveal the Currently frame. Which features a single scrollbox that feeds directly into "CU" (like the profile currently box). Input is sanitzed and version incremented like normal.
image

There's also a chat command "/trp3 currently" which opens the same chat frame. In case the toolbar is not used or you want to hide the button.
image

There are three new strings that will require localization.

### CODE REVIEW

File: totalRP3/Modules/Dashboard/Dashboard.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@ https://raw.githubusercontent.com/Meorawr/wow-ui-schema/main/UI.xsd">
 	<Include file="HTMLContent.xml"/>
 	<Include file="TabFrame.xml"/>
 	<Include file="StatusPanel.xml"/>
+	<Include file="Currently.xml" />
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Dashboard.xml:18]:
```suggestion
	<Include file="Currently.xml"/>
```

File: totalRP3/Modules/Dashboard/Currently.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<!--
+	Copyright The Total RP 3 Authors
+	SPDX-License-Identifier: Apache-2.0
+-->
+<Ui xmlns="http://www.blizzard.com/wow/ui/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.blizzard.com/wow/ui/
+https://raw.githubusercontent.com/Meorawr/wow-ui-schema/main/UI.xsd">
+
+	<Include file="Currently.lua"/>
+
+	<Frame name="TRP3_CURRENTLY" inherits="TRP3_AltHoveredFrame" enableMouse="true" hidden="true" movable="true" clampedToScreen="true" parent="UIParent" frameStrata="DIALOG">
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.xml:10]:
```suggestion
	<Frame name="TRP3_CurrentlyFrame" inherits="TRP3_AltHoveredFrame" enableMouse="true" hidden="true" movable="true" clampedToScreen="true" parent="UIParent" frameStrata="DIALOG">
```
WHY ARE WE SHOUTING OUR FRAME NAMES

File: totalRP3/Modules/Dashboard/Currently.lua [hunk 3]
```diff
@@ -0,0 +1,59 @@
+-- Copyright The Total RP 3 Authors
+-- SPDX-License-Identifier: Apache-2.0
+
+TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOADED, function()
+	local loc = TRP3_API.loc;
+	local setupMovableFrame = TRP3_API.ui.frame.setupMove;
+
+	local frame = TRP3_CURRENTLY;
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.lua:8]:
```suggestion
	local frame = TRP3_CurrentlyFrame;
```

**Review by Meorawr — APPROVED** [review]

**Review by Meorawr — CHANGES_REQUESTED** [review]
also do the thing

File: totalRP3/Modules/Dashboard/Currently.xml (hunk 2 above)
Comment (Raenore) [reply totalRP3/Modules/Dashboard/Currently.xml:10]: YES

**Comment by Raenore** [comment]
> also do the thing

Yes.

File: totalRP3/Modules/Dashboard/Currently.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		<Layers>
+			<Layer level="OVERLAY">
+				<FontString parentKey="title" inherits="GameFontNormalLarge" justifyH="CENTER" justifyV="MIDDLE" setAllPoints="true">
+					<Size x="0" y="30"/>
+					<Anchors>
+						<Anchor point="TOP" x="0" y="-10"/>
+						<Anchor point="RIGHT" x="-10" y="0"/>
+						<Anchor point="LEFT" x="10" y="0"/>
+					</Anchors>
+					<Color r="0.95" g="0.95" b="0.95"/>
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.xml:29]:
```suggestion
				<FontString parentKey="Title" inherits="GameFontHighlightLarge" justifyH="CENTER" justifyV="MIDDLE" setAllPoints="true">
					<Size x="0" y="30"/>
					<Anchors>
						<Anchor point="TOP" x="0" y="-10"/>
						<Anchor point="RIGHT" x="-10" y="0"/>
						<Anchor point="LEFT" x="10" y="0"/>
					</Anchors>
```

File: totalRP3/Modules/Dashboard/Currently.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+					<KeyValue key="styleName" value="CloseButton" type="string"/>
+				</KeyValues>
+				<Anchors>
+					<Anchor point="TOPRIGHT" x="-5" y="-5"/>
+				</Anchors>
+				<Scripts>
+					<OnClick function="HideParentPanel"/>
+				</Scripts>
+			</Button>
+			<Frame parentKey="currentlyText" inherits="TRP3_TextArea">
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.xml:46]:
```suggestion
			<Frame parentKey="CurrentlyText" inherits="TRP3_TextArea">
```

File: totalRP3/Modules/Dashboard/Currently.xml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+				<Size y="144"/>
+				<Anchors>
+					<Anchor point="TOPLEFT" relativePoint="BOTTOMLEFT" relativeKey="$parent.title" x="11" y="6"/>
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.xml:49]:
```suggestion
					<Anchor point="TOPLEFT" relativePoint="BOTTOMLEFT" relativeKey="$parent.Title" x="11" y="6"/>
```

File: totalRP3/Modules/Dashboard/Currently.lua [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	local function showCurrentlyFrame()
+		frame.currentlyText.scroll.text:SetText(AddOn_TotalRP3.Player.GetCurrentUser():GetCurrentlyText());
+
+		frame:Show();
+	end
+	TRP3_API.r.showCurrentlyFrame = showCurrentlyFrame;
+
+	frame.title:SetText(loc.CURRENTLY_TITLE);
+
+	frame.currentlyText.scroll.text:HookScript("OnTextChanged", onCurrentlyChanged);
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.lua:30]:
```suggestion
	frame.Title:SetText(loc.CURRENTLY_TITLE);
	frame.CurrentlyText.scroll.text:HookScript("OnTextChanged", onCurrentlyChanged);
```

File: totalRP3/Modules/Dashboard/Currently.lua [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		if not userInput then
+			return;
+		end
+
+		AddOn_TotalRP3.Player.GetCurrentUser():SetCurrentlyText(editBox:GetText());
+	end
+
+	-- Prepares to show frame, gets player's "currently" and writes it to the scrollbox.
+	local function showCurrentlyFrame()
+		frame.currentlyText.scroll.text:SetText(AddOn_TotalRP3.Player.GetCurrentUser():GetCurrentlyText());
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Currently.lua:22]:
```suggestion
		frame.CurrentlyText.scroll.text:SetText(AddOn_TotalRP3.Player.GetCurrentUser():GetCurrentlyText());
```

**Review by Meorawr — CHANGES_REQUESTED** [review]
I was going to be nice and not point out the inconsistencies but nooooooooooooooooooooooooooooooooo solly has to bring it up

**Review by Meorawr — APPROVED** [review]
big pog

**Pull request merged**